client/my-sites/comments/controller.js

```diff
@@ -12,27 +12,57 @@ import route from 'lib/route';
 import CommentsManagement from './main';
 import controller from 'my-sites/controller';
 
-export const comments = function( context ) {
-	const siteSlug = route.getSiteFragment( context.path );
-	const status = context.params.status === 'pending' ? 'unapproved' : context.params.status;
+export const isStatusValid = status => {
+	const validStatuses = [ 'pending', 'approved', 'spam', 'trash', 'all' ];
+	return validStatuses.indexOf( status ) !== -1;
+};
+
+export const getRedirect = ( status, domain ) => {
+	const statusValidity = isStatusValid( status );
+	if ( status === domain ) {
+		return '/comments/pending/' + domain;
+	}
+	if ( ! statusValidity && ! domain ) {
+		return '/comments';
+	}
+	if ( ! statusValidity && domain ) {
+		return '/comments/pending/' + domain;
+	}
+	if ( statusValidity && ! domain ) {
+		return '/comments/' + status;
+	}
+	return false;
+};
+
+export const comments = function( context, next ) {
+	const { status } = context.params;
+	const domain = route.getSiteFragment( context.path );
+	const redirect = getRedirect( status, domain );
+
+	if ( redirect ) {
+		return page.redirect( redirect );
+	}
+
+	controller.navigation( context, next );
 
 	renderWithReduxStore(
 		<CommentsManagement
 			basePath={ context.path }
-			siteSlug={ siteSlug }
-			status={ status }
+			siteSlug={ domain }
+			status={ 'pending' === status ? 'unapproved' : status }
 		/>,
 		'primary',
 		context.store
 	);
 };
 
-export const sites = function( context ) {
+export const sites = function( context, next ) {
 	const { status } = context.params;
-	const siteSlug = route.getSiteFragment( context.path );
+	const domain = route.getSiteFragment( context.path );
```

Comment:
are we sure this is a domain and not a site slug? there are cases where we get special characters in the slug or fragment that aren't the domain: Jetpack sites I think